pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotBrokerDebug.java

```diff
@@ -158,7 +158,7 @@ public Map<ServerInstance, List<String>> getRoutingTableForQuery(
       if (!_accessControlFactory.create()
           .hasAccess(httpHeaders, TargetType.TABLE, brokerRequest.getQuerySource().getTableName(),
               Actions.Table.GET_ROUTING_TABLE)) {
-        throw new WebApplicationException("Permission denied", Response.Status.FORBIDDEN);
+        throw new WebApplicationException("Unauthorized", Response.Status.UNAUTHORIZED);
```

Comment:
Let's keep 403 here, and handle the authN issues in the access control object, such as for BasicAuthAccessControl, the change should be done [HERE](https://github.com/apache/pinot/blob/70c4c5b8019b3c3b71865bde71e4899c3659082e/pinot-broker/src/main/java/org/apache/pinot/broker/broker/BasicAuthAccessControlFactory.java#L90) .
Instead of
      if (!principalOpt.isPresent()) {
        // no matching token? reject
        return false;
      }

try:

      if (!principalOpt.isPresent()) {
        // no matching token? reject
       throw new WebApplicationException("No token", Response.Status.UNAUTHORIZED);
      }

Reply:
Can you please take a look? Thanks